packages/@lightningjs/ui-components/src/mixins/withThemeStyles/style-manager.test.js

```diff
@@ -0,0 +1,326 @@
+      styleManager._addCache(name, newPayload);
+      const cacheKey = styleManager._generateCacheKey(name);
+      const themeCache = cache.get(cacheKey);
+      expect(themeCache).toEqual({
+        ids: expectedIds,
+        payload: newPayload
+      });
+    });
+  });
+
+  describe('_cleanupCache', () => {
+    it('should remove the current component id from the cache entries', () => {
+      const name1 = 'styleSource';
+      const name2 = `style_${component.mode}_${component.tone}`;
+      const existingIds = ['123', component.__id, '456'];
+      const expectedIds = ['123', '456'];
+      const payload = { color: 'red' };
+      cache.set(styleManager._generateCacheKey(name1), {
+        ids: existingIds,
+        payload
+      });
+      cache.set(styleManager._generateCacheKey(name2), {
+        ids: existingIds,
+        payload
+      });
+      styleManager._cleanupCache();
+      const cacheKey1 = styleManager._generateCacheKey(name1);
+      const cache1 = cache.get(cacheKey1);
+      const cacheKey2 = styleManager._generateCacheKey(name2);
+      const cache2 = cache.get(cacheKey2);
+      expect(cache1).toEqual({
+        ids: expectedIds,
+        payload
+      });
+      expect(cache2).toEqual({
+        ids: expectedIds,
+        payload
+      });
+    });
+
+    it('should remove the cache entry if the current component id is the only id in the entry', () => {
+      const name = 'styleSource';
+      const existingIds = [component.__id];
+      const payload = { color: 'red' };
+      cache.set(styleManager._generateCacheKey(name), {
+        ids: existingIds,
+        payload
+      });
+      styleManager._cleanupCache();
+      const cacheKey = styleManager._generateCacheKey(name);
+      const themeCache = cache.get(cacheKey);
+      expect(themeCache).toBeUndefined();
+    });
+  });
+
+  describe('_removeCache', () => {
+    it('should remove the cache entry with the given name', () => {
+      const name = 'styleSource';
+      const payload = { color: 'red' };
+      const cacheKey = styleManager._generateCacheKey(name);
+      cache.set(cacheKey, {
+        ids: [component.__id],
+        payload
+      });
+      styleManager._removeCache(cacheKey);
+      const themeCache = cache.get(cacheKey);
+      expect(themeCache).toBeUndefined();
+    });
+  });
+
+  describe('_getCache', () => {
+    it('should return the cache entry with the given name', () => {
+      const name = 'styleSource';
+      const payload = { color: 'red' };
+      const cacheKey = styleManager._generateCacheKey(name);
+      cache.set(cacheKey, {
+        ids: [component.__id],
+        payload
+      });
+      const themeCache = styleManager._getCache(name);
+      expect(themeCache).toEqual({
+        ids: [component.__id],
+        payload
+      });
+    });
+  });
+
+  describe('_update', () => {
+    it.skip('should update the component style based on changes in tone, mode, and _componentLevelStyle', () => {
```

Comment:
If these are going to stay skipped, is it worth throwing in a comment for these about why they're being skipped so that we know where to pick up with them eventually?

Reply:
Removing these tests for now. I will have a fast follow PR to add more in depth unit tests for this feature.